### ISSUE TITLE
setuptools.setup function does not warn on wrong user keywords

### ISSUE THREAD

**Issue by maersdal** [pypa/setuptools#2188]
The setup function does not warn or fail when called with wrong user parameters

repro: take ANY valid setup.py file, change the 'name' parameter to 'names' and do a `pip install .`

this will cause pip to install UNKNOWN.version 

Suggested fix:
have a checker on setup. Before calling _install_setup_requires, which checks the parameters, with good warning messages.

**Comment by Kixunil** [comment]
Yeah, this would be great, bit me today (`extras_require` vs `extras_requires`).

**Comment by Boole215** [comment]
I'm going to be taking a look at this issue/trying to get a checker working
